OpenRA.Mods.Common/Traits/RevealsMap.cs

```diff
@@ -0,0 +1,78 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Reveals shroud and fog across the whole map while active.")]
+	public class RevealsMapInfo : ConditionalTraitInfo
+	{
+		[Desc("Stance the watching player needs to see the shroud removed.")]
+		public readonly Stance ValidStances = Stance.Ally;
+
+		[Desc("Can this actor reveal shroud generated by the `GeneratesShroud` trait?")]
+		public readonly bool RevealGeneratedShroud = true;
+
+		public override object Create(ActorInitializer init) { return new RevealsMap(this); }
+	}
+
+	public class RevealsMap : ConditionalTrait<RevealsMapInfo>, INotifyKilled, INotifyActorDisposing
+	{
+		readonly RevealsMapInfo info;
```

Comment:
`ConditionalTrait<RevealsMapInfo>.Info` is a `RevealsMapInfo`, so this would never be needed. Use `Info` instead.

Reply:
```suggestion
```